packages/mui-system/src/createStyled.js

```diff
@@ -191,10 +194,48 @@ export default function createStyled(input = {}) {
       shouldForwardPropOption = undefined;
     }
 
-    const defaultStyledResolver = styledEngineStyled(tag, {
-      shouldForwardProp: shouldForwardPropOption,
+    if (shouldForwardPropInput) {
+      shouldForwardPropOption = shouldForwardPropInput;
+    }
+
+    // Always propagated the sx prop, the augmented component will make sure to intercept if needed
+    const shouldForwardPropEmotionOption = (prop) =>
+      prop === 'sx' ||
+      shouldForwardPropOption === undefined ||
+      (typeof shouldForwardPropOption === 'function' && shouldForwardPropOption(prop));
+
+    // This is needed in order to intercept the transformed sx values potentially done by the zero-runtime library
+    const AugmentedTag = React.forwardRef(function Component(props, ref) {
+      const { sx, ...other } = props;
+      const sxClass = typeof sx === 'string' ? sx : sx?.className;
+      const sxVars = sx && typeof sx !== 'string' ? sx.vars : undefined;
+      const sxVarsStyles = {};
+
+      if (sxVars) {
```

Comment:
This logic prevent use from not being able to just use the sx style function. Another alternative would be adding a useSx hook in all components. Alternatively we maybe could use an existing hook, like useThemeProps to avoid changes in all components.

Reply:
Maybe we can update the styleSxFunction, see https://github.com/mui/material-ui/blob/6528bfb1b2a885b0b096c0f097b2d198305b8bc8/packages/mui-system/src/styleFunctionSx/styleFunctionSx.js#L80

Reply:
The styleFunctionSx can only return a style object, it can't process the sx prop, this is why it was disregarded as a solution.